### ISSUE TITLE
Please make it easier to write comments

### ISSUE THREAD

**Issue by arnonax** [ploeh/ploeh.github.com#827]
Writing a comment on your excellent blog posts is very combersome. It starts with finding the relevant article in GitHub (after I was already at the blog post itself), goes through finding the correct place and syntax to add the comment, editing the comment as markup, and then going through the process of creating a PR...

At the very least, a direct link from each post to its corresponding GitHub file (in edit mode) would be very beneficial.

Consider that I (and I bet that many others) usually read your posts from a smartphone, going through this process really turns me down from taking the effort to write comments.

Thanks in advance,
Arnon.

**Comment by arnonax** [comment]
You can phrase it something like this:
"To leave a comment on this post according to [these]() guidelines, click [here]().” (the links here are dummy).

This way you still point the readers to the guidelines while make it easy to go directly to editing if you're already familiar with them.

IMHO this would definitely be an improvement.

**Comment by SeanKilleen** [comment]
Happened to stumble across this while considering making a comment.

You might be interested in using [Giscus](https://giscus.app/) for comments. I've really enjoyed it so far.

The basic premise is that it uses (a locked-down category of) GitHub discussions. Someone logs in via their GitHub account, and is able to post on the related discussion (one topic per thread). Includes preview of markdown formatting, configurable ability to add reactions, etc.

What I like the most is that it keeps the content on GitHub, and I'm not forced to rely on a 3rd party application to hold the data. I'm already all-in on hosting via GitHub, so it seemed a natural extension. Some balk at requiring a GitHub account, but I figure that offers me some ecosystem-level protection against bad actors while lowering the bar for being able to make a comment.

If you're interested in this, I'd be happy to take a whack at implementing it in your Jekyll blog, as a way of saying thanks for the great content over the years.

**Comment by ploeh** [comment]
@SeanKilleen, thank you for writing. Suggestions like that comes up from time to to. Please see [my answer to one of them](https://github.com/ploeh/ploeh.github.com/issues/661#issuecomment-698429830).

For reference, here are a few similar suggestions:

- #808
- #747
- #618

**Comment by SeanKilleen** [comment]
@ploeh fair enough. [Your comment above from July](https://github.com/ploeh/ploeh.github.com/issues/827#issuecomment-1192327442) seemed to indicate your main concerns were:

* "someone would still need to know the formatting rules" (solved by the UI of Giscus and similar tools) 
* "leaving such a link without further instructions would invite more arbitrary editing of the files" (not a problem when using tools like Giscus which have a nice UI around them).

Your other points in your links are of course fine trade-offs to choose; it's your space. For me it ends up being about "barrier to entry" level for comments. I felt I might have had something to contribute, but skipped it because I didn't feel like direct editing a file in a custom way and submitting a pull request (given that the link at the bottom of the post about comments doesn't link to the file I'd need to edit). That may be entirely in line with how you feel about comments on your blog and you may even find it to be an effective filter for anything other than highly-motivated commenters who understand PRs.  👍 

---

If you are interested, I'd be happy to submit a PR to update the text and make the comment link an edit link for the file where someone can propose a PR. If not, no worries and I'll consider this closed on my end.

**Comment by ploeh** [comment]
> If you are interested, I'd be happy to submit a PR to update the text and make the comment link an edit link for the file where someone can propose a PR. If not, no worries and I'll consider this closed on my end.

Thank you, that's an interesting experiment. I'm happy to try it, but I'll roll it back if it turns out that it generates too much noise.
